job.go

```diff
@@ -0,0 +1,253 @@
+package jasper
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1"
+	"errors"
+	"fmt"
+
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/level"
+	"github.com/mongodb/grip/send"
+)
+
+type amboyJob struct {
```

Comment:
Is there a reason that `amboyJob` has to have a bunch of duplicate `CreateOptions` and `OutputOptions` settings instead of just having a `CreateOptions` field?

Reply:
It makes it possible for the constructors to not return an error, and defer that to runtime, which I think makes some things easier, and I think flattening the object will make a bit more sense in the database.